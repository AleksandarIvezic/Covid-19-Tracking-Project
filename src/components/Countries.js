import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Country from './Country';
import Search from './Search';
import '../styling/Countries.css';

const Countries = ({
  countries, total, loading, handleClick,
}) => {
  const [filteredCountries, setFilteredCountries] = useState(Object.keys(countries));

  useEffect(() => {
    setFilteredCountries(Object.keys(countries));
  }, [countries]);

  const filterCountries = (text) => Object.keys(countries).filter((country) => {
    const regex = new RegExp(text, 'gi');
    return country.match(regex);
  });
  const handleChange = (e) => setFilteredCountries(filterCountries(e.target.value));

  return (
    <div>
      <div className="header d-flex p-1">
        <img className="c-img" src="World_map_svg.svg" alt="world-map" width="50%" height="50%" />
        <div className="right-side d-flex flex-column align-items-center justify-content-center w-50 fw-bold">
          <h1 className="m-0 fs-3 lh-1 fw-bolder">WORLD</h1>
          <span className="lh-1">
            {total && Intl.NumberFormat('de-DE').format(total.today_confirmed)}
          </span>
          <span className="lh-1">total cases</span>
        </div>
      </div>
      <Search handleChange={handleChange} />
      {loading && <img className="loading-img" src="virus-img.png" alt="loader" />}
      <ul className="d-flex flex-wrap">
        {filteredCountries && filteredCountries.map((key) => (
          (key !== 'Kosovo')
          && (
            <li key={key} className="w-50">
              <Country
                country={key}
                totalConfirmed={countries[key].today_confirmed}
                handleClick={handleClick}
              />
            </li>
          )
        ))}
      </ul>
      {!filteredCountries.length
      && !loading
      && <h3>SORRY, WE DON&apos;T HAVE RESULTS FOR YOUR SERCH!!! :(</h3>}
    </div>
  );
};

Countries.defaultProps = {
  loading: true,
  total: {},
};

Countries.propTypes = {
  countries: PropTypes.objectOf(PropTypes.object).isRequired,
  total: PropTypes.shape({
    today_confirmed: PropTypes.number,
  }),
  loading: PropTypes.bool,
  handleClick: PropTypes.func.isRequired,
};

export default Countries;
