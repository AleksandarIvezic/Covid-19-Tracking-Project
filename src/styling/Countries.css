.header {
  background: #5687e4;
}

li:nth-child(4n+0) {
  background-color: #4b74c2;
}

li:nth-child(4n+1) {
  background-color: #4b74c2;
}

li:nth-child(4n+2) {
  background-color: #3e61a3;
}

li:nth-child(4n+3) {
  background-color: #3e61a3;
}

.map {
  transform: scale(1);
}
