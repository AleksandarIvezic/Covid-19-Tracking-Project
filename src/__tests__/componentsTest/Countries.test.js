import React from 'react';
import renderer from 'react-test-renderer';
import Countries from '../../components/Countries';

const countriesMock = {
  Afganistan: {
    new_cases: 23,
    deaths: 2,
    today_confirmed: 23432,
  },
  Albania: {
    new_cases: 13,
    new_deaths: 0,
    today_confirmed: 23432,
  },
};

const totalMock = {
  today_confirmed: 23432,
  today_deaths: 4878813,
  today_new_confirmed: 439801,
};

describe('Countries', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Countries
          countries={countriesMock}
          total={totalMock}
          loading={false}
          handleClick={() => 2 + 3}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
