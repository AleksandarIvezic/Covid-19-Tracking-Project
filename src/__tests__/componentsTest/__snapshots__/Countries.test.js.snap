// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders correctly 1`] = `
<div>
  <div
    className="header d-flex p-1"
  >
    <img
      alt="world-map"
      className="c-img"
      height="50%"
      src="World_map_svg.svg"
      width="50%"
    />
    <div
      className="right-side d-flex flex-column align-items-center justify-content-center w-50 fw-bold"
    >
      <h1
        className="m-0 fs-3 lh-1 fw-bolder"
      >
        WORLD
      </h1>
      <span
        className="lh-1"
      >
        23.432
      </span>
      <span
        className="lh-1"
      >
        total cases
      </span>
    </div>
  </div>
  <form
    className="d-flex stripe-text justify-content-between"
  >
    <span
      className="m-1"
    >
      STATS BY COUNTRY
    </span>
    <input
      className="search"
      name="search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </form>
  <ul
    className="d-flex flex-wrap"
  >
    <li
      className="w-50"
    >
      <button
        className="country position-relative country text-light"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="country"
          className="c-img"
          height="100"
          src="/World_map_svg.svg"
          width="100"
        />
        <div
          className="bottom d-flex justify-content-end"
        >
          <div
            className="text d-flex flex-column justify-content-center"
          >
            <span
              className="fw-bold text-uppercase "
            >
              Afganistan
            </span>
            <span>
              23.432
            </span>
          </div>
        </div>
        <i
          className="far fa-arrow-alt-circle-right position-absolute top-0 end-0 p-2"
        >
           
        </i>
      </button>
    </li>
    <li
      className="w-50"
    >
      <button
        className="country position-relative country text-light"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="country"
          className="c-img"
          height="100"
          src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/al/vector.svg"
          width="100"
        />
        <div
          className="bottom d-flex justify-content-end"
        >
          <div
            className="text d-flex flex-column justify-content-center"
          >
            <span
              className="fw-bold text-uppercase "
            >
              Albania
            </span>
            <span>
              23.432
            </span>
          </div>
        </div>
        <i
          className="far fa-arrow-alt-circle-right position-absolute top-0 end-0 p-2"
        >
           
        </i>
      </button>
    </li>
  </ul>
</div>
`;
